test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/amendment/group/ConfirmRemoveMemberControllerSpec.scala

```diff
@@ -0,0 +1,152 @@
+
+import java.util.UUID
+
+class ConfirmRemoveMemberControllerSpec
+    extends ControllerSpec with MockAmendmentJourneyAction with TableDrivenPropertyChecks {
+
+  private val mcc  = stubMessagesControllerComponents()
+  private val page = mock[confirm_remove_member_page]
+
+  private val sessionId = UUID.randomUUID().toString
+
+  private val controller =
+    new ConfirmRemoveMemberController(mockAuthAllowEnrolmentAction,
+                                      mockAmendmentJourneyAction,
+                                      mcc,
+                                      page
+    )
+
+  private def authorisedUserWithPptSubscription() =
+    authorizedUser(user =
+      newUser().copy(enrolments =
+        Enrolments(
+          Set(
+            new Enrolment(PptEnrolment.Identifier,
+                          Seq(EnrolmentIdentifier(PptEnrolment.Key, "XMPPT0000000123")),
+                          "activated"
+            )
+          )
+        )
+      )
+    )
+
+  val anotherGroupMember                     = groupMember.copy(id = "another-group-member-id")
+  private val groupMembers: Seq[GroupMember] = Seq(groupMember, anotherGroupMember)
+
+  val populatedGroupDetails =
+    GroupDetail(membersUnderGroupControl = Some(true), members = groupMembers)
+
+  private val populatedRegistrationWithGroupMembers =
+    aRegistration().copy(groupDetail = Some(populatedGroupDetails))
+
+  override protected def beforeEach(): Unit = {
+    inMemoryRegistrationAmendmentRepository.reset()
+    reset(mockSubscriptionConnector)
+    simulateGetSubscriptionSuccess(populatedRegistrationWithGroupMembers)
+    when(page.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.raw("Confirm remove member"))
+  }
+
+  "Confirm Remove Member Controller" when {
+    "signed in user has a ppt registration to amend" should { // TODO what if it isn't a group registration?
+      "show page" in {
+        authorisedUserWithPptSubscription()
+        simulateGetSubscriptionSuccess(populatedRegistrationWithGroupMembers)
+
+        val resp =
+          controller.displayPage(groupMember.id)(getRequest())
+
+        status(resp) mustBe OK
+      }
+
+      "update registration" when {
+        "signed in user when a ppt registration requests removal of an existing group member" in {
+          authorisedUserWithPptSubscription()
+          simulateGetSubscriptionSuccess(populatedRegistrationWithGroupMembers)
+
+          // postRequestEncoded will not encode a yes form correctly
+          val correctlyEncodedForm = Seq("value" -> "yes")
+          val resp = controller.submit(groupMember.id)(
+            postRequestTuplesEncoded(correctlyEncodedForm, continueFormAction, sessionId)
+          )
+
+          status(resp) mustBe SEE_OTHER
+          val updatedReg = await(inMemoryRegistrationAmendmentRepository.get(sessionId)).get
+          updatedReg.groupDetail.get.members.size mustBe 1
+          updatedReg.groupDetail.get.members.head mustBe groupMember
```

Comment:
We should probably also verify that the subscription has been updated. This is only verifying that the local repository was updated.

Reply:
That `mockAmendmentJourneyAction` isn't itself a mock might be alittle unfair here =)